File: Linq2TwitterDemos_Console/BlockDemos.cs
using System;
using System.Linq;
using System.Threading.Tasks;
using LinqToTwitter;

namespace Linq2TwitterDemos_Console
{
    class BlockDemos
    {
        internal static async Task RunAsync(TwitterContext twitterCtx)
        {
            char key;

            do
            {
                ShowMenu();

                key = Console.ReadKey(true).KeyChar;

                switch (key)
                {
                    case '0':
                        Console.WriteLine("\n\tListing Blocked Users...\n");
                        await ListBlockedUsersAsync(twitterCtx);
                        break;
                    case 'q':
                    case 'Q':
                        Console.WriteLine("\nReturning...\n");
                        break;
                    default:
                        Console.WriteLine(key + " is unknown");
                        break;
                }

            } while (char.ToUpper(key) != 'Q');
        }

        static async Task ListBlockedUsersAsync(TwitterContext twitterCtx)
        {
            var blockResponse =
                await
                    (from block in twitterCtx.Blocks
                     where block.Type == BlockingType.List
                     select block)
                    .SingleOrDefaultAsync();

            blockResponse.Users.ForEach(user => 
                Console.WriteLine(user.ScreenNameResponse));
        }

        static void ShowMenu()
        {
            Console.WriteLine("\nBlock Demos - Please select:\n");

            Console.WriteLine("\t 0. List Blocked Users");
            Console.WriteLine();
            Console.WriteLine("\t Q. Return to Main menu");
        }
    }
}

File: Samples/LinqToTwitter6/Console/ConsoleDemos.VisualBasic/ConsoleDemos.VisualBasic/BlockDemos.vb
Imports LinqToTwitter

Friend Class BlockDemos
    Friend Shared Async Function RunAsync(twitterCtx As TwitterContext) As Task
        Dim blockResponse =
            Await _
                (From block In twitterCtx.Blocks
                 Where block.Type = BlockingType.List
                 Select block) _
                .SingleOrDefaultAsync()

        If blockResponse IsNot Nothing And blockResponse.Users IsNot Nothing Then
            blockResponse.Users.ForEach(
                Sub(user) Console.WriteLine(user.ScreenNameResponse))
        End If
    End Function
End Class
